test/passing/tests/quotations.ml.js-ref

```diff
@@ -0,0 +1,121 @@
+#syntax quotations on
+
+let simple_number = <[123]>
+
+let npower x_quoted n =
+  let rec loop m = if m = 0 then <[1]> else <[$x_quoted * $(loop (m - 1))]> in
+  loop n
+;;
+
+let example_function n = <[fun x -> $(npower <[x]> n)]>
+
+let longer_example m n =
+  let first_quote = <[fun x -> $(npower <[x]> m)]> in
+  let second_quote = <[fun y -> $(npower <[y]> n)]> in
+  let combined_quote = <[fun x -> $second_quote ($first_quote x)]> in
+  <[$combined_quote, $combined_quote]>
+;;
+
+let even_longer m n =
+  <[fun x y ->
+    let xm = $(npower <[x]> m) in
+    let yn = $(npower <[y]> n) in
+    let z = xm + yn in
+    $(npower <[z]> (m + n))]>
+;;
+
+type s = <[int]>
+type t = s expr
+
+let f (x : t) : <[$s * $s]> expr = <[$x, $x + 1]>
+
+let double =
+  <[let x = <[42]> in
+    <[123 + $x]>]>
+;;
+
+(* Long lines and breaks *)
+
+let _ =
+  <[let aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa = 1 in
+    [ aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    ; aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    ; aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    ]]>
+;;
+
+let _ =
+  <[fun xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
+      zzzzzzzzzzzzzzzzzzzzzzz ->
+    $( <[xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+         + yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy]>
+     , <[zzzzzzzzzzzzzzzzzzzzzzz]> )]>
+;;
+
+let _ =
+  let xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx =
+    <[fun aaaaaaaaaaaaaaaaaaaaa -> aaaaaaaaaaaaaaaaaaaaa, aaaaaaaaaaaaaaaaaaaaa]>
+  in
+  <[( $xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+    , $xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+        (fun bbbbbbbbbbbbbbbbbbbbb -> bbbbbbbbbbbbbbbbbbbbb) )]>
+;;
+
+let _ =
+  fun xxxxxxxxxxxxxxxxxxxx ->
+  <[fun zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz ->
+    $((fun yyyyyyyyyyyyyyyyyyyy -> <[zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz + 42]>)
+        xxxxxxxxxxxxxxxxxxxx)]>
+;;
+
+(* Comments *)
+
+(* let q = <[fun xxxxxxxxxx -> (xxxxxxxxxx, xxxxxxxxxx, xxxxxxxxxx)]> in let f a =
+   <[$q $a + $q $a + $q $a]> in f <[111111111111111111]> *)
+
+let _ =
+  <[2222222222222222222222222222222222
```

Comment:
I think this would look nicer if there were a space between the bracket and the inner term in the cases where the innter term gets split to multiple lines - this is what we do for tuple and lists brackets, for example.  I think this can be done with `wrap_fits_break` or similar functions.